delta-app/src/main/java/io/cdap/delta/store/RemoteStateStore.java

```diff
@@ -68,17 +67,17 @@ public void writeOffset(DeltaWorkerId id, OffsetAndSequence offset) {
   @Override
   public void writeState(DeltaWorkerId id, String key, byte[] val) {
     String valString = new String(val, StandardCharsets.UTF_8);
-    assessmentServiceClient.retryableApiCall(getMethodPathState(id, key), HttpMethod.PUT, valString, null);
+    assessmentServiceClient.retryableApiCall(getMethodPathState(id, key), HttpMethod.PUT, valString, String.class);
```

Comment:
As commented before, do not try to represent a byte array as a String. It is incorrect. For example, there are some byte arrays that are invalid UTF8 Strings.